src/compiler/checker.ts

```diff
@@ -2989,12 +2983,9 @@ namespace ts {
         }
 
         function typePredicateToString(typePredicate: TypePredicate, enclosingDeclaration?: Declaration, flags?: TypeFormatFlags): string {
-            const writer = getSingleLineStringWriter();

```

Comment:
I take it this is the only place that `getSingleLineStringWriter` was used? I wonder why it had the complex pooling behaviour then. Do you know whether it was used in more locations in the past?